src/main/java/net/minestom/server/item/event/ItemEventRegistry.java

```diff
@@ -0,0 +1,45 @@
+package net.minestom.server.item.event;
+
+import net.minestom.server.item.ItemTag;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+
+public class ItemEventRegistry<T> {
+
+    private final HashMap<T, ItemEventIdentifierHandler<T>> identifierHandlerHashMap = new HashMap<>();
+    private final ItemTag<T> tag;
+
+    ItemEventRegistry(ItemTag<T> tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Finds an ItemEventIdentifierHandler by the identifier value, and creates a new instance if none were found.
+     *
+     * @param identifier The identifier used to find the ItemEventIdentifierHandler
+     *
+     * @return The found ItemEventIdentifierHandler with a new instance if none were found.
+     */
+    public @NotNull ItemEventIdentifierHandler<T> identifierOrNew(T identifier) {
+        if (identifierHandlerHashMap.containsKey(identifier)) return identifierHandlerHashMap.get(identifier);
```

Comment:
Couldn't it be replaced by `Map#putIfAbsent`?

Reply:
Map#putIfAbsent's return value is null, meaning the check still has to occur resulting in essentially the same code.

Reply:
Moved to Map#computeIfAbsent :+1: